salt/states/btrfs.py

```diff
@@ -119,8 +119,8 @@ def wrapper(*args, **kwargs):
             kwargs['__dest'] = dest
             ret = action(*args, **kwargs)
         except Exception as e:
-            log.error('''Traceback: {}'''.format(traceback.format_exc()))
-            ret['comment'].append(e)
+            log.exception('Encountered error mounting %s', device)
+            ret['comment'].append(six.text_type(e))
```

Comment:
You introduced two error here:

* This error is not a mount error, is the exception produced in the method that is wrapping this code (the call from line 120). The mount error was taken care a bit before.

* 'e' do not contain the traceback, and we do not want to hide it. We need to print it in the log at least (and maybe in the comments too)

I will change that in a different pull request, and I will integrate your fix for log.exception and six.text_type

Reply:
>     * This error is not a mount error, is the exception produced in the method that is wrapping this code (the call from line 120). The mount error was taken care a bit before.

OK, so the message in the log was inaccurate, fair enough.

>     * 'e' do not contain the traceback, and we do not want to hide it. We need to print it in the log at least (and maybe in the comments too)

I have no idea what you're talking about. Nothing was hidden. Logging at `exception` level is the same thing as doing `log.error('some message', exc_info=True)`. It includes the exception.